sklearn/datasets/openml.py

```diff
@@ -0,0 +1,306 @@
+        datasets with the same name.
+
+    version : integer or 'active', default='active'
+        Version of the dataset. Can only be provided if also ``name`` is given.
+        If 'active' the oldest version that's still active is used.
+
+    id : int
+        OpenML ID of the dataset. The most specific way of retrieving a
+        dataset. If ID is not given, name (and potential version) are
+        used to obtain a dataset.
+
+    data_home : string or None, default None
+        Specify another download and cache folder for the data sets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
+
+    target_column_name : string or None, default 'default-target'
+        Specify the column name in the data to use as target. If
+        'default-target', the standard target column a stored on the server
+        is used. If ``None``, all columns are returned as data and the
+        target is ``None``.
+
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    Returns
+    -------
+
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
+        'data', the data to learn, 'target', the regression target or
+        classification labels, 'DESCR', the full description of the dataset,
+        'feature_names', the original names of the dataset columns, and
+        'details' which provide more information on the openml meta-data.
+    """
+    data_home = get_data_home(data_home=data_home)
+    data_home = join(data_home, 'openml')
+    if cache:
+        mem = Memory(join(data_home, 'cache'), verbose=0).cache
+    else:
+        def mem(func):
+            return func
+    _get_data_info_by_name_ = mem(_get_data_info_by_name)
+    _get_data_description_by_id_ = mem(_get_data_description_by_id)
+    _get_data_features_ = mem(_get_data_features)
+    _download_data_ = mem(_download_data)
+
+    if not exists(data_home):
+        os.makedirs(data_home)
+
+    # check legal function arguments. id XOR (name, version) should be provided
+    if name is not None:
+        if id is not None:
+            raise ValueError(
+                "Dataset id={} and name={} passed, but you can only "
+                "specify a numeric id or a name, not both.".format(id, name))
+        data_info = _get_data_info_by_name_(name, version)
+        data_id = data_info['did']
+    elif id is not None:
+        # from the previous if statement, it is given that name is None
+        if version is not "active":
+            raise ValueError(
+                "Dataset id={} and version={} passed, but you can only "
+                "specify a numeric id or a version, not both.".format(id,
+                                                                      name))
+        data_id = id
+    else:
+        raise ValueError(
+            "Neither name nor id are provided. Please provide name xor id.")
```

Comment:
xor -> or (I know that is was here on purpose, but think that many people don't know xor)

Reply:
Agreed